actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -32,5 +44,13 @@ def method_for_action(action_name)
         "default_render"
       end
     end
+
+    private
+      def no_templates?
+        # Check and see if any templates exist for the given action name, and prefixes.
+        # Variants are provided also as an argument, to check in case a template exists
+        # for a format not specifically declared in the `respond_to` block.
+        !template_exists_in_any_format?(action_name.to_s, _prefixes, { variants: request.variant })
```

Comment:
It would fail if only the action name and prefix were passed as arguments, because since the format is JSON, it matters that whatever being returned is coming from a JSON template. Otherwise, just the XML (or whatever) would be returned. It's possible that the method name is what's confusing -- I'm all ears to change it if there's a name that's easier to understand.

Also, I tried swapping out the new method in `no_templates?` with the already existing `template_exists?` method, and that caused the test above to fail.